### PR TITLE
[OCPCLOUD-1809]: Update tooling for MAO

### PR DESCRIPTION
**Pull request by odvarkadaniel**
This commit introduces modernization of the approach to update auxiliary tooling to be more consistent such as in Control Plane Machine Set repository. We align `envtest`, `controller-gen` and other tooling as well as in the CPMS repo.

### CODE REVIEW

**Comment by damdo** [comment]
Looks like some lines you added are already there: https://github.com/openshift/machine-api-operator/pull/1100/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R55-R58

**Comment by damdo** [comment]
Is not on all our repo yet, we started with the CPMSO and are moving them one at a time. Looks like this one was already done by @lobziik here: https://github.com/openshift/machine-api-operator/pull/1071

**Comment by odvarkadaniel** [comment]
> Is not on all our repo yet, we started with the CPMSO and are moving them one at a time. Looks like this one was already done by @lobziik here: #1071

I see. I was wondering why there was inconsistency in the way we use `$KUBEBUILDER_ASSETS` in `unit` instead of `test` here: https://github.com/openshift/machine-api-operator/pull/1100#issuecomment-1370999606
I suggest to place the `$KUBEBUILDER_ASSETS` is `test` as we do in CPMS, wdyt?

File: hack/ci-test.sh [hunk 1]
```diff
@@ -23,6 +23,8 @@ REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
 OPENSHIFT_CI=${OPENSHIFT_CI:-""}
 ARTIFACT_DIR=${ARTIFACT_DIR:-""}
 
+TEST_RESULT=$?
+
 function runTestsCI() {
```
Comment (JoelSpeed) [inline hack/ci-test.sh:28]: Lets make sure to get this script updated to match the CPMS one please

File: Makefile [hunk 2]
```diff
@@ -6,6 +6,12 @@ IMAGE        = $(REGISTRY)machine-api-operator
 BUILD_IMAGE ?= registry.ci.openshift.org/openshift/release:golang-1.18
 GOLANGCI_LINT = go run ./vendor/github.com/golangci/golangci-lint/cmd/golangci-lint
 
+# ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
+ENVTEST_K8S_VERSION = 1.25
+
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+ENVTEST = go run ${PROJECT_DIR}/vendor/sigs.k8s.io/controller-runtime/tools/setup-envtest
```
Comment (JoelSpeed) [inline Makefile:13]: These are duplicated below

Comment (odvarkadaniel) [reply Makefile:13]: Do we want to have `1.26` version here as well?

Comment (JoelSpeed) [reply Makefile:13]: Yeah I think lets try getting all of the envtest updated to 1.26, and other deps if they need to as well

**Comment by odvarkadaniel** [comment]
The latest commit bumps k8s envtest to version `1.26`. There are also minor changes to some of the files due to the version changes in vendor packages. 
We also use `make test` as  a wrapper for `make unit`. I find this to be convenient for the future additions so we can wrap some more stuff into `make test` as we do in [CPMS' Makefile](https://github.com/openshift/cluster-control-plane-machine-set-operator/blob/main/Makefile#L90) repository.

**Comment by odvarkadaniel** [comment]
There is an issue with a packages `github.com/openshift/library-go/pkg/operator/v1helpers` and `k8s.io/client-go/tools/cache`.
`library-go` package is dependent on `client-go` with struct `fakeSharedInformer` (can be seen here: https://github.com/openshift/library-go/blob/master/pkg/operator/v1helpers/test_helpers.go#L23). With changes in `client-go` that break this (this file: https://github.com/kubernetes/client-go/blob/master/tools/cache/shared_informer.go#L557-L559). That new addition is that the function `AddEventHandler` from `client-go` now returns `(ResourceEventHandlerRegistration, error)`, which is not compatible with the code from `library-go` package, so we can never compile our code. 
I can either downgrade the `k8s.io` packages back to `1.25` or do something else. What would you suggest to do in this situation, @damdo, @JoelSpeed?

**Comment by JoelSpeed** [comment]
@odvarkadaniel your issue with library go will be resolved by https://github.com/openshift/library-go/pull/1442 which is updating the function within library-go, perhaps we can reach out to find out what the status of that PR is?

**Comment by odvarkadaniel** [comment]
I have updated the Go version to `1.19` and also the `envtest `version is bumped to `1.26`. In order for this to be working, we need to wait to the PR that @JoelSpeed mentioned in the comment above.

File: pkg/controller/machineset/machineset_controller_suite_test.go [hunk 3]
```diff
@@ -56,7 +56,7 @@ var (
 func TestMachinesetController(t *testing.T) {
 	RegisterFailHandler(Fail)
 
-	RunSpecs(t, "MachineSet Controller Suite")
+	RunSpecs(t, "Cluster Operator status controller Suite")
```
Comment (JoelSpeed) [inline pkg/controller/machineset/machineset_controller_suite_test.go:59]: Why has this been changed?

File: pkg/controller/vsphere/machineset/controller_suite_test.go [hunk 4]
```diff
@@ -45,7 +45,7 @@ var (
 func TestReconciler(t *testing.T) {
 	RegisterFailHandler(Fail)
 
-	RunSpecs(t, "Controller Suite")
+	RunSpecs(t, "Cluster Operator status controller Suite")
```
Comment (JoelSpeed) [inline pkg/controller/vsphere/machineset/controller_suite_test.go:48]: Why has this been changed?

File: pkg/controller/machineset/machineset_controller_suite_test.go (hunk 3 above)
Comment (odvarkadaniel) [reply pkg/controller/machineset/machineset_controller_suite_test.go:59]: I don't remember tbh, but this change came in commit 57805a53802149c35b51ca1e305165739092e202

File: pkg/controller/vsphere/machineset/controller_suite_test.go (hunk 4 above)
Comment (odvarkadaniel) [reply pkg/controller/vsphere/machineset/controller_suite_test.go:48]: Same as the mention above.

File: pkg/controller/machineset/machineset_controller_suite_test.go (hunk 3 above)
Comment (JoelSpeed) [reply pkg/controller/machineset/machineset_controller_suite_test.go:59]: I think we can change the names back, they don't need to be changed

Comment (odvarkadaniel) [reply pkg/controller/machineset/machineset_controller_suite_test.go]: Resolved.

**Comment by odvarkadaniel** [comment]
This PR only introduced changes to the way we run unit tests. The PR #1115 is responsible for the bump of k8s deps.

File: Makefile (hunk 2 above)
Comment (JoelSpeed) [reply Makefile:13]: This is still duplicating existing vars below, can you please double check what you've added here in a wider context

**Pull request merged**